From dfe88d95fb5358186d68129efb0df6753a2a89e7 Mon Sep 17 00:00:00 2001
From: Marc Mao <marc.mao@intel.com>
Date: Tue, 14 Feb 2023 18:55:49 +0800
Subject: [WA] Fix crash in texcompress_compute destroy

texcompress_compute will not be initialzed for GLES1.x context because it
doesn't support compute shader extension. But texcompress_compute is still
be called in the h/w astc patch.

Change-Id: I8c3aae6c501e6f86d1b4108c795b451b50316647
Signed-off-by: Marc Mao <marc.mao@intel.com>
---
 src/mesa/state_tracker/st_texcompress_compute.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/mesa/state_tracker/st_texcompress_compute.c b/src/mesa/state_tracker/st_texcompress_compute.c
index a2f68d91960..cceadc12f67 100644
--- a/src/mesa/state_tracker/st_texcompress_compute.c
+++ b/src/mesa/state_tracker/st_texcompress_compute.c
@@ -780,6 +780,9 @@ destroy_astc_decoder(struct st_context *st)
 void
 st_destroy_texcompress_compute(struct st_context *st)
 {
+   if (!st->texcompress_compute.progs)
+      return;
+
    /* The programs in the array are part of the gl_context (in st->ctx).They
     * are automatically destroyed when the context is destroyed (via
     * _mesa_free_context_data -> ... -> free_shader_program_data_cb).
-- 
2.33.1

